apps/storefront/src/pages/InvoicePayment/index.test.tsx

```diff
@@ -0,0 +1,244 @@
+
+      when(getCreateCartResponse)
+        .calledWith(stringContainingAll('{invoiceId: 3344, amount: "433" }', 'currency: "USD"'))
+        .thenReturn({
+          data: {
+            invoiceCreateBcCart: {
+              result: {
+                cartId: 'foo-bar',
+                checkoutUrl: 'https://testing.com/checkout?cartId=foo-bar',
+              },
+            },
+          },
+        });
+
+      renderWithProviders(<Payment />, {
+        preloadedState: { company: loggedInUser, global: globalCatalyst },
+      });
+
+      await waitFor(() => {
+        expect(window.location.href).toEqual('http://localhost/checkout?cartId=foo-bar');
+      });
+    });
+  });
+
+  describe('when rendered in "other"', () => {
+    const globalOther = buildGlobalStateWith({ storeInfo: { platform: 'other' } });
+
+    it('navigates to checkout using a login-redirect link', async () => {
+      const getInvoiceDetails = vi.fn();
+      const getCheckoutLoginResponse = vi.fn();
+      const getCreateCartResponse = vi.fn();
+
+      vi.mocked(useParams).mockReturnValue({ id: '3344' });
+
+      server.use(
+        graphql.query('GetInvoiceDetails', ({ query }) =>
+          HttpResponse.json(getInvoiceDetails(query)),
+        ),
+        graphql.mutation('CreateCart', ({ query }) =>
+          HttpResponse.json(getCreateCartResponse(query)),
+        ),
+        graphql.mutation('checkoutLogin', ({ variables }) =>
+          HttpResponse.json(getCheckoutLoginResponse(variables)),
+        ),
+      );
+
+      when(getInvoiceDetails)
+        .calledWith(stringContainingAll('invoiceId: 3344'))
+        .thenReturn({ data: { invoice: { openBalance: { code: 'USD', value: '433' } } } });
+
+      when(getCreateCartResponse)
+        .calledWith(stringContainingAll('{invoiceId: 3344, amount: "433" }', 'currency: "USD"'))
+        .thenReturn({
+          data: {
+            invoiceCreateBcCart: {
+              result: { cartId: 'foo-bar', checkoutUrl: faker.internet.url() },
+            },
+          },
+        });
+
+      when(getCheckoutLoginResponse)
+        .calledWith({ cartData: { cartId: 'foo-bar' } })
+        .thenReturn({
+          data: {
+            checkoutLogin: {
+              result: { redirectUrl: 'https://example.com/checkout?cartId=foo-bar' },
+            },
+          },
+        });
+
+      renderWithProviders(<Payment />, {
+        preloadedState: { company: loggedInUser, global: globalOther },
+      });
+
+      await waitFor(() => {
+        expect(window.location.href).toEqual('https://example.com/checkout?cartId=foo-bar');
+      });
+    });
+  });
+
+  it('shows an `invoice can not be blank error`', async () => {
+    vi.mocked(useParams).mockReturnValue({ id: '' });
+
+    renderWithProviders(<Payment />, { preloadedState: { company: loggedInUser } });
+
+    await waitFor(() => {
+      expect(screen.getByText('The invoice cannot be blank')).toBeInTheDocument();
```

Comment:
feels more like a 404 scenario?

Though changing it is more of a feature improvement than we'd consider now